fastlane_core/lib/fastlane_core/PTY.rb

```diff
@@ -0,0 +1,22 @@
+# rubocop:disable Style/FileName
+# via https://github.com/fastlane/fastlane/pull/7202/files
+module FastlaneCore
+  class PTY
+    def self.spawn(*cmd, &block)
+      # Utopia - we have PTY
+      if Gem::Specification.any? { |s| s.name == "pty" }
+        require "pty"
+        PTY.spawn(command) do |stdout, stdin, pid|
+          block.call(stdin, stdout, pid)
+        end
+      else
+        # Sucks - but lets try to handle it
+        require "systemu"
+        stdout = ''
+        stderr = ''
+        _status = systemu cmd, 'stdout' => stdout, 'stderr' => stderr
```

Comment:
is `systemu`  added to the gemspec? how'd we handle that, for users that do not have systemu installed?!

Reply:
best approach would be to check if gem is installed and if not output a nice fastlane-style error on how to obtain the dependency.

Reply:
> is systemu added to the gemspec? how'd we handle that, for users that do not have systemu installed?!

Reply:
> is systemu added to the gemspec? how'd we handle that, for users that do not have systemu installed?!

Good catch, I added it to fastlane.gemspec as you did in the PR where I copied this code from.
I went by the existing code for `pty` that also doesn't appear in `fastlane.gemspec` and doesn't have any additional error handling if it doesn't exist.

Reply:
> is systemu added to the gemspec? how'd we handle that, for users that do not have systemu installed?!

Good catch, I added it to fastlane.gemspec as you did in the PR where I copied this code from.

Fastlane-style error handling is a bit above my ruby knowledge tbh. Feel free to submit into my PR though if you want to implement it, otherwise I think we should be fine for now.

Reply:
ohhh - i  i think you missunderstood me - adding a hard dep. (as on my old PR) should be avoided, i guess.

How about 

```
Actions.verify_gem!('systemu')
require "systemu"
```

then you'll get a nice error message, and installation instructions, + now new hard dependency (just because we support windows)

Reply:
Ah, now I got. Changed (and also made the comments a bit nicer)